### PR TITLE
:wrench: Refactor button delete fix #109

### LINKED ISSUE #109: Warning message before action patient delete

### PR DESCRIPTION
**Pull request by issa-diallo**
# Description of the feature 
A confirmation delete popup which is going to be visible when the user wants to delete a todo.

# Technical solution
- I created a Modal component that takes a state as parameter
> https://github.com/issa-diallo/mynotif_frontend/issues/109

### CODE REVIEW

File: src/components/ModalDelete.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+const ModalDelete: FunctionComponent<ModalProps> = ({ handleClose, show, onDelete }) => {
+  return (
+    <>
+      <Modal show={show} onHide={handleClose}>
+        <Modal.Header>
+          <Modal.Title>Confirmation</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>Etes vous sur de vouloir supprimer ce patient ?</Modal.Body>
+        <Modal.Footer>
+          <Button variant='danger' onClick={() => handleClose(false)}>
```
Comment (AndreMiras) [inline src/components/ModalDelete.tsx:21]: I'm not sure `handleClose()` takes a boolean parameter judging from the rest of the code

File: src/components/ModalDelete.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            Retour
+          </Button>
+          <Button variant='primary' onClick={onDelete}>
+            Supprimer
+          </Button>
```
Comment (AndreMiras) [inline src/components/ModalDelete.tsx:26]: I think the color code should be almost the opposite. The delete button should be red and the cancel should not be red

**Review by AndreMiras — COMMENTED** [review]
Whatever we put in the pull request body should be in the commit message.
Since we're spending time writing explanations, why not making it in the commit for later reference?

**Comment by AndreMiras** [comment]
How about the prescription delete?
I think it's fine to address only one at a time, but let's make the #109 task more specific and create a follow up task for the other entries we can delete

File: src/components/ModalDelete.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { FunctionComponent } from 'react'
+
+import Button from 'react-bootstrap/Button'
+import Modal from 'react-bootstrap/Modal'
+
+interface ModalProps {
+  handleClose: any
+  show: boolean
+  onDelete: any
```
[the remark this answers is not in the document]
Comment (issa-diallo) [reply src/components/ModalDelete.tsx]: fix

File: src/components/ModalDelete.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { FunctionComponent } from 'react'
+
+import Button from 'react-bootstrap/Button'
+import Modal from 'react-bootstrap/Modal'
+
+interface ModalProps {
```
[the remark this answers is not in the document]
Comment (issa-diallo) [reply src/components/ModalDelete.tsx]: modify

File: src/components/ModalDelete.tsx (hunk 1 above)
Comment (issa-diallo) [reply src/components/ModalDelete.tsx]: exactly I had fix

File: src/components/ModalDelete.tsx (hunk 2 above)
Comment (issa-diallo) [reply src/components/ModalDelete.tsx]: fix

**Review by AndreMiras — APPROVED** [review]

**Pull request merged**